packages/app/src/App.vue

```diff
@@ -1,35 +1,74 @@
 <template>
   <template v-if="isReady">
-    <the-header :class="$route?.name" />
-    <div class="container-app">
+    <the-header v-if="!isPublicRoute" :class="route?.name" />
+    <div
+      :class="{
+        'container-app': !isPublicRoute,
+      }"
+    >
       <IndexerDelayAlert v-if="!currentNetwork.maintenance && currentNetwork.name === 'mainnet'" />
       <MaintenanceView v-if="currentNetwork.maintenance" />
       <router-view v-else />
     </div>
-    <the-footer />
+    <the-footer v-if="!isPublicRoute" />
   </template>
 </template>
 
 <script setup lang="ts">
+import { onMounted, ref, watchEffect } from "vue";
+import { useRoute, useRouter } from "vue-router";
+
 import { useTitle } from "@vueuse/core";
 
+import useEnvironmentConfig from "./composables/useEnvironmentConfig";
+import usePublicRoutes from "./composables/usePublicRoutes";
+
 import IndexerDelayAlert from "@/components/IndexerDelayAlert.vue";
 import TheFooter from "@/components/TheFooter.vue";
 import TheHeader from "@/components/header/TheHeader.vue";
 
 import useContext from "@/composables/useContext";
 import useLocalization from "@/composables/useLocalization";
+import useLogin from "@/composables/useLogin";
+import usePrividiumRpc from "@/composables/usePrividiumRpc";
 import useRouteTitle from "@/composables/useRouteTitle";
 
 import MaintenanceView from "@/views/MaintenanceView.vue";
 
 const { setup } = useLocalization();
 const { title } = useRouteTitle();
+const context = useContext();
+const { prividium } = useEnvironmentConfig();
+const route = useRoute();
+const router = useRouter();
+const { isPublicRoute } = usePublicRoutes();
+const { initializeLogin } = useLogin(context);
+const { initializePrividiumRpcUrl } = usePrividiumRpc();
 
 useTitle(title);
 const { isReady, currentNetwork } = useContext();
 
 setup();
+
+const isAuthCheckComplete = ref(false);
+
+onMounted(async () => {
+  if (prividium.value) {
```

Comment:
Seems like this code only works if networks are set via runtime config. We should either make it work with environment config or get rid of environment config.